### DISCUSSION TITLE
How to deploy in Digital Ocean

### DISCUSSION THREAD

**Discussion by lazyneto** [aahnik/tgcf#111] [Q&A] [answered]
I tried to deploy a app in digital ocean, but when i build it i get this message in console:

```
unknown field "forwards"
```

My config https://prnt.sc/12eh39z

```shell

Deploy Error: Non-Zero Exit Code
Your deploy failed because your container exited with a non-zero exit code.
Error code: DeployContainerExitNonZero

[2021-05-02T19:47:48.512350746Z] WARNING:root:Configuration not found! Learn more http://bit.ly/configure-tgcf
```

**Comment by aahnik** [comment]
Please give more details. You must have made some error.

Have you set environment variables correctly ?

**Reply by aahnik** [reply]
hi, @crowOcean, can't you read this line? 

**WARNING:root: Configuration not found! Learn more http://bit.ly/configure-tgcf**

-----

Haven't you read the wiki?

You need to create an environment variable named `TGCF_CONFIG` that contains the configuration.

If you are using an Ubuntu Droplet, or running `tgcf` on you own computer, you may use the `tgcf.config.yml` file to write the configuration. In places where you can't create a file, you should use the environment variable.

**Reply by lazyneto** [reply]
I can read, since I am not blind ... the wiki does not mention these steps, update your documentation to be more clear and objective, not everyone has the knowledge to read source code and try to guess what is needed

**Reply by aahnik** [reply]
hi, @crowOcean i am extremely sorry for missing this point. i had written it in [readme earlier](https://github.com/aahnik/tgcf/tree/v0.1.17#run-on-cloud-%EF%B8%8F), but removed, it from there, thinking I will add to wiki.

sorry, missed it.

will update the docs soon.

thanks for using and giving feedback

**Reply by lazyneto** [reply]
Np :)

**Comment by aahnik** [comment, marked as answer]
Some details in the wiki were missing. The **wiki pages have been updated.**

Read
- https://github.com/aahnik/tgcf/wiki/Environment-Variables
- https://github.com/aahnik/tgcf/wiki/How-to-configure-tgcf-%3F
- https://github.com/aahnik/tgcf/wiki/Deploy-to-Digital-Ocean
